### PR TITLE
[IOTDB-4085] Add StateWindowAccessStrategy in UDF

### PR DESCRIPTION
**Pull request by ZhanGHanG9991**
## Description
Add StateWindowAccessStrategy in UDF

<hr>

This PR has:
- [x] been self-reviewed.
    - [ ] concurrent read
    - [ ] concurrent write
    - [ ] concurrent read and write 
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [x] added or updated version, __license__, or notice information
- [x] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [x] added integration tests.
- [x] been tested in a test IoTDB cluster.

<hr>

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/intermediate/SingleInputColumnSingleReferenceIntermediateLayer.java [hunk 1]
```diff
@@ -530,4 +532,204 @@ public RowWindow currentWindow() {
...
+      }
+
+      @Override
+      public RowWindow currentWindow() {
+        return window;
+      }
+    };
+  }
+
+  boolean splitWindowForStateWindow(
```
Comment (lancelly) [inline server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/intermediate/SingleInputColumnSingleReferenceIntermediateLayer.java:653]: Duplicate code. Please move it to TransformUtils.

**Review by lancelly — CHANGES_REQUESTED** [review]
PTAL

File: server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/intermediate/SingleInputColumnSingleReferenceIntermediateLayer.java (hunk 1 above)
Comment (ZhanGHanG9991) [reply server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/intermediate/SingleInputColumnSingleReferenceIntermediateLayer.java]: I've moved it to TransformUtils.

**Review by lancelly — APPROVED** [review]
LGTM

**Pull request merged**